### PR TITLE
fix: Popovers in Explore not attached to the fields they are triggered by

### PR DESCRIPTION
**Pull request by diegomedina248**
### SUMMARY
Popovers are attached not to the triggers, but rather the overlay on the top of the page. This results in a few problems:
* Pointers are not always accurate and do not point to the field they were triggered by
* when user is scrolling through the control panel, the popover does not move and is floating in random places

The solution does two things:
* Blocks scrolling in the panel container while the popup is opened
* Calculates the popover placement based on the trigger location in the screen.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF
Before:

https://user-images.githubusercontent.com/17252075/158194589-3ed951fe-5000-43ff-8630-c31bfe287769.mov

After:

https://user-images.githubusercontent.com/17252075/158194993-33140bfc-d3e6-4e6b-bd4f-d90753dd6770.mov

### ADDITIONAL INFORMATION

- [ ] Has associated issue:
- [ ] Required feature flags:
- [x] Changes UI
- [ ] Includes DB Migration (follow approval process in [SIP-59](https://github.com/apache/superset/issues/13351))
  - [ ] Migration is atomic, supports rollback & is backwards-compatible
  - [ ] Confirm DB migration upgrade and downgrade tested
  - [ ] Runtime estimates and downtime expectations provided
- [ ] Introduces new feature or API
- [ ] Removes existing feature or API

### CODE REVIEW

**Review by rusackas — APPROVED** [review]
Well done! This looks great. The only quirk I noticed is that if you have a popover open and shrink the actual window (not a common occurrence I'm worried about) the popover becomes misaligned. 

This could be fixed by using a debounced function on window resize to call calculatePlacement. But as I said, not too worried about this.

Approving, but also cc-ing @kgabryje since he knows these popovers and their fragilities quite well.

File: superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterPopoverTrigger/AdhocFilterPopoverTrigger.test.tsx [hunk 1]
```diff
@@ -73,7 +73,10 @@ test('should be visible when controlled', async () => {
       Click
     </AdhocFilterPopoverTrigger>,
   );
-  expect(screen.getByRole('tooltip')).toBeInTheDocument();
+
+  await waitFor(() => {
```
Comment (kgabryje) [inline superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterPopoverTrigger/AdhocFilterPopoverTrigger.test.tsx:77]: I think we can replace `waitFor + getByRole` with `findByRole`

**Review by kgabryje — APPROVED** [review]
Amazing work! That's been an issue for SUCH a long time, I'm super happy to see it fixed. Approving, I just left 1 minor comment (feel free to ignore it if the CI passes 😄)

File: superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterPopoverTrigger/AdhocFilterPopoverTrigger.test.tsx (hunk 1 above)
Comment (diegomedina248) [reply superset-frontend/src/explore/components/controls/FilterControl/AdhocFilterPopoverTrigger/AdhocFilterPopoverTrigger.test.tsx]: Agreed, thanks!

**Pull request merged**